pkg/util/filesystem/util_windows.go

```diff
@@ -88,6 +90,160 @@ func IsUnixDomainSocket(filePath string) (bool, error) {
 	return true, nil
 }
 
+// On Windows os.Mkdir all doesn't set any permissions so call the Chown function below to set
+// permissions once the directory is created.
+func MkdirAll(path string, perm os.FileMode) error {
+	klog.V(6).InfoS("Function MkdirAll starts", "path", path, "perm", perm)
+	err := os.MkdirAll(path, perm)
+	if err != nil {
+		return fmt.Errorf("Error creating directory %s: %v", path, err)
+	}
+
+	err = Chmod(path, perm)
+	if err != nil {
+		return fmt.Errorf("Error setting permissions for directory %s: %v", path, err)
+	}
+
+	return nil
+}
+
+const (
+	// These aren't defined in the syscall package for Windows :(
+	USER_READ      = 0x100
+	USER_WRITE     = 0x80
+	USER_EXECUTE   = 0x40
+	GROUP_READ     = 0x20
+	GROUP_WRITE    = 0x10
+	GROUP_EXECUTE  = 0x8
+	OTHERS_READ    = 0x4
+	OTHERS_WRITE   = 0x2
+	OTHERS_EXECUTE = 0x1
+	USER_ALL       = USER_READ | USER_WRITE | USER_EXECUTE
+	GROUP_ALL      = GROUP_READ | GROUP_WRITE | GROUP_EXECUTE
+	OTHERS_ALL     = OTHERS_READ | OTHERS_WRITE | OTHERS_EXECUTE
+)
+
+// On Windows os.Chmod only sets the read-only flag on files, so we need to use Windows APIs to set the desired access on files / directories.
+// The OWNER mode will set file permissions for the file owner SID, the GROUP mode will set file permissions for the file group SID,
+// and the OTHERS mode will set file permissions for BUILTIN\Users.
+// Please note that Windows containers can be run as one of two user accounts; ContainerUser or ContainerAdministrator.
+// Containers run as ContainerAdministrator will inherit permissions from BUILTIN\Administrators,
+// while containers run as ContainerUser will inherit permissions from BUILTIN\Users.
+// Windows containers do not have the ability to run as a custom user account that is known to the host so the OTHERS group mode
+// is used to grant / deny permissions of files on the hosts to the ContainerUser account.
+func Chmod(path string, filemode os.FileMode) error {
```

Comment:
it is *very* surprising to me to see this here instead of in stdlib or x/sys/windows ... go *really* expects all callers to reimplement this themselves?

Reply:
The stdlib's version of Chmod does note that only on Windows it only marks files are owner writable or not
https://pkg.go.dev/os#Chmod and the x/sys/windows package has the same behavior - https://cs.opensource.google/go/x/sys/+/refs/tags/v0.22.0:windows/syscall_windows.go;l=799

I was _very_ surprised to find that os.MkdirAll( ) just igores passed in file permissions entirely on Windows 
https://cs.opensource.google/go/x/sys/+/refs/tags/v0.22.0:windows/syscall_windows.go;l=667-673


The approach I took here to grant file permissions on Windows is a little bit opinionated by the fact that Windows containers running as `ContainerAdminstrator` inherit permissions from `BUILTIN\Administrators`.

Windows file permissions are a lot more nuanced (you can add explicit allow and deny permissions, more gradual permissions for each security identifier, etc) and I don't think the goland community was ever able to reach consensus on how to handle this in os or x/sys/windows :-/

Reply:
ugh! thanks for the context @marosset